setup.sh

```diff
@@ -150,54 +173,75 @@ else
-                        apply $2
-                else
-                        if [ "${ENV}" = 'staging' -o "${ENV}" = 'production' ] ; then
-                                echo "Cannot run terraform apply all on ${ENV}"
+                does_stack_config_exist
+                if [ $? = 1 ]    ; then
+                        if [ $2 ] ; then
+                                apply $2
                         else
-                                for folder in ${COMPONENTS[@]}
-                                do 
-                                        apply $folder
-                                done
+                                if [ "${ENV}" = 'staging' -o "${ENV}" = 'production' ] ; then
+                                        echo "Cannot run terraform apply all on ${ENV}"
+                                else
+                                        for folder in ${COMPONENTS[@]}
+                                        do
+                                                apply $folder
+                                        done
+                                fi
                         fi
                 fi
         ;;
         -d) echo "Destroy terraform plan to environment: ${ENV}"
-                if [ "${ENV}" = 'staging' -o "${ENV}" = 'production' ] ; then
-                        echo "Cannot run terraform apply all on ${ENV}"
-                else
-                        read -p 'Are you sure? (y)' answer
-
-                        if [ "${answer}" = 'y' ] ; then
-                                for folder in ${COMPONENTSDESTROY[@]}
-                                do
-                                        destroy $folder
-                                done
+                does_stack_config_exist
+                if [ $? = 1 ] ; then
+                        if [ $2 ] ; then
+                                destroy $2
+                        else
+                                if [ "${ENV}" = 'staging' -o "${ENV}" = 'production' ] ; then
+                                        echo "Cannot run terraform destroy all on ${ENV}"
+                                else
+                                        read -p 'Are you sure? (yN)' answer
+
+                                        if [ "${answer}" = 'y' ] ; then
+                                                for folder in ${COMPONENTSDESTROY[@]}
+                                                do
+                                                        destroy $folder
+                                                        if [ $? != 0 ] ; then
+                                                                exit
+                                                        fi
+                                                done
+                                        fi
+                                fi
                         fi
                 fi
         ;;
         *) echo "Invalid option"
         ;;
         esac
 fi
+
+cd $ROOTPROJ
```

Comment:
I don't think this will do anything? [scripts are run inside a subshell, so changing directory in the subshell doesn't affect the caller](https://stackoverflow.com/a/255415)

Reply:
ok will remove